### DISCUSSION TITLE
[suggestion] Add classes to TAGs, Posts & Pages

### DISCUSSION THREAD

**Discussion by raramuridesign** [GetPublii/Publii#1759] [General] [answered]
Hi All
Publii allows one to add a class to the menu item, but there is no option to add a class to the following areas.

1. Tags - ie when you create a tag
2. Pages
3. Posts

If the class could be added into the body element that would be perfect.
If we had the possibility to add classes to each of these it would really be useful. 

For Tags specifically, we would like to hide the intro on a certain tag, there is no option to do this on the tag settings (which would be another great addition).

Hope to get some feedback on this.
M

**Comment by bobmitro** [comment]
In the latest theme updates (related to the pages), the following classes were added to other body elements:

```
home-template
tag-template
tags-template
page-template
post-template
author-template
pagination-template
error-template
search-template
```

If you want more control, you can add more classes by editing the theme files, especially the **head.hbs**

**Comment by raramuridesign** [comment]
@bobmitro Yes, I am aware of the new classes, and these work well.
But adding a custom class based on a tag would be super useful.  So If I have several tags and would like to hide certain elements on those pages, currently I have to add javascript to add a unique class to the page identified by the url, it would be better to use a class on the page from the TAG setup. 

Does this make sense ?
M

**Reply by bobmitro** [reply, marked as answer]
It makes sense, but you can apply it yourself. I don’t want it to be like the WP, where many classes are added to the site.

**Comment by raramuridesign** [comment]
Thanks @bobmitro Understood. I will continue with my current approach.

**Reply by candidexmedia** [reply]
I'm a few months late, but this discussion might be useful if you're still looking for a solution:

https://github.com/GetPublii/Publii/discussions/1847

The doc ref for Handlebars tags for Publii tag pages is here: https://getpublii.com/dev/tag-tags/

You could also consider creating custom templates for your tags (see: https://forum.getpublii.com/topic/how-to-add-custom-tag-template/)
